### PR TITLE
update contributing doc with better workflow

### PR DESCRIPTION
**Pull request by jxu-flexport**
by using `bindfs` we can have changes from the latitude repo update in realtime to the monorepo and also switch between local latitude and the `yarn.lock` bundled version instantly (vs waiting for yarn upgrade)

symlinks did not play well with flow (flow resolved the symlinks to the linked location so everything was messed up because it thought everything was under `../latitude` instead of `node_modules/latitude`).

More people should test this out before this is merged as the de facto set up

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,25 +24,28 @@ You don't need to wait for a Latitude PR to be merged before beginning your
...
+brew cask install osxfuse
 ```
 
-After pushing updates to your latitude branch, run `yarn upgrade latitude` from
-your monorepo to see your changes there. Feel free to push as many times as you
-like. Remember, you can squash changes before opening a PR or requesting review.
+To set this up, point your monorepo's latitude dependency to your local repo:
+
+```bash
+bindfs -r <full_path_to_latitude_repo> <full_path_to_monorepo>/node_modules/latitude
```
Comment (jxu-flexport) [inline CONTRIBUTING.md]: `-r` flag provides a safety feature where the mirror folder is read-only and you can't accidentally change it like if you run `yarn install` with the mirror still mounted

**Comment by dmnd** [comment]
TIL about `bindfs`. I haven't tried it yet, but this sounds badass! Thanks for the suggestion.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,25 +24,28 @@ You don't need to wait for a Latitude PR to be merged before beginning your
...
-To set this up, point your monorepo's latitude dependency to your branch:
+We will use the following tools to help sync the local repos:
 
 ```bash
-cd ~/monorepo
-
-# point to your latitude feature branch
-yarn upgrade latitude@https://github.com/flexport/latitude.git#feature-branch-name
+brew install bindfs
+brew cask install osxfuse
```
Comment (Cytancy) [inline CONTRIBUTING.md]: NIT: Order should be reversed, osxfuse must be installed before bindfs can.

**Comment by Cytancy** [comment]
This is a great tip, we are currently looking into improving the overall Latitude contribution workflow ([LDS-531](https://flexport.atlassian.net/browse/LDS-531)). I was able to get `bindfs` working successfully without issue.

We currently we use [`yarn link](https://yarnpkg.com/lang/en/docs/cli/link/)` to do the same behavior:

To link:
`cd location/of/latitude`
`yarn link`
`cd location/of/flexport`
`yarn link latitude`

To unlink:
`cd location/of/flexport`
`yarn unlink latitude`
OR
`cd location/of/latitude`
`yarn unlink`

I will close this for now in favor of the above since it requires no additional packages since it comes out of the box with yarn but this above `yarn link` flow should be mentioned in `CONTRIBUTING.md`.

**Comment by jxu-flexport** [comment]
@Cytancy one problem I was having with `yarn link` was that it broke `flow` typechecks because it treated files under the symlink as `<symlink_dir>/file` instead of `node_modules/latitude/file`. I don't think either one is going to be the long-term solution though

**Comment by Cytancy** [comment]
@jxu-flexport Ah, I am sorry about that, I run into the same issue and usually resolve it by restarting flow. Agreed this isn't this best solution, we're considering tools such as FBShipIt to make the process seamless, any suggestions would be appreciated.

**Comment by Cytancy** [comment]
Reopening, having both approaches is valid and would be good to keep track of them in `CONTRIBUTING.md` since the flow issue can be annoying. @notandrewkaye @ddzoan

**Pull request merged**